http_service/bugbug_http/app.py

```diff
@@ -85,24 +91,73 @@ class UnauthorizedError(Schema):
     message = fields.Str(default="Error, missing X-API-KEY")
 
 
+class BranchName(Schema):
+    branch = fields.Str(example="autoland")
+
+
+class Schedules(Schema):
+    tasks = fields.List(fields.Str)
+    groups = fields.List(fields.Str)
+
+
+class PushNotFoundError(Schema):
+    message = fields.Str(default="Error: specified push not found!")
+
+
 spec.components.schema(BugPrediction.__name__, schema=BugPrediction)
-spec.components.schema(
-    BugPredictionNotAvailableYet.__name__, schema=BugPredictionNotAvailableYet
-)
+spec.components.schema(NotAvailableYet.__name__, schema=NotAvailableYet)
 spec.components.schema(ModelName.__name__, schema=ModelName)
 spec.components.schema(UnauthorizedError.__name__, schema=UnauthorizedError)
+spec.components.schema(BranchName.__name__, schema=BranchName)
+spec.components.schema(Schedules.__name__, schema=Schedules)
+spec.components.schema(PushNotFoundError.__name__, schema=PushNotFoundError)
 
 
 api_key_scheme = {"type": "apiKey", "in": "header", "name": "X-API-Key"}
 spec.components.security_scheme("api_key", api_key_scheme)
 
 
+def get_target_func():
+    if "predict" in request.path:
+        return classify_bug
+
+    if request.path.startswith("/push"):
+        return schedule_tests
+
+    raise ValueError(f"Invalid request: {request.path}")
+
+
 def get_job_id():
     return uuid.uuid4().hex
 
 
-def get_mapping_key(model_name, bug_id):
-    return f"bugbug:mapping_{model_name}_{bug_id}"
+def get_mapping_key(*args):
+    """Return a key that can be used to map `args` to a job id.
+
+    If args aren't specified, a key that maps `request.path` to a job id will
+    be used instead.
+    """
+    func = get_target_func()
+    return f"bugbug:job_id:{func.__name__}:{'_'.join(args)}"
```

Comment:
What if we just use the args? Since the model name is in there, we shouldn't have problems

Reply:
> I think it would be better if we implemented a JobInfo dataclass that stored all the job related state (rather than passing func and *args around all over the place). I started implementing it and then realized it was distracting me more than it should, so decided to leave it for a follow up commit.

And yes, this would probably look better, but we can leave it for a follow-up.

Reply:
It would work for now, but technically args aren't unique across jobs. The unique identifier would be function + args. Imagine we add a `/commit/<rev>/schedules` API in the future. Then a `<rev>` could be both a valid argument to the `/push` and the `/commit` endpoints. At that point, we'd need the function name to help uniquely identify those jobs. So I think it's best to leave it in.

I guess we could remove the `bugbug:job_id` prefix, but I figured it might help if we ever needed to inspect the redis store manually. I'm also not sure what other types of keys are being placed in there, so it helps avoid conflicts in general.